policy-controller/k8s/index/src/outbound/index/unmeshed_network.rs

```diff
@@ -0,0 +1,230 @@
+pub(crate) struct Networks(pub Vec<Cidr>);
+
+#[derive(Debug, Default)]
+pub(crate) struct UnmeshedNetwork {
+    pub networks: Networks,
+    pub name: String,
+    pub namespace: String,
+    pub creation_timestamp: Option<DateTime<Utc>>,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+struct MatchedUnmeshedNetwork {
+    matched_cidr: MatchedCidr,
+    name: String,
+    namespace: String,
+    creation_timestamp: Option<DateTime<Utc>>,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+struct MatchedCidr(Cidr);
+
+// === impl MatchedUnmeshedNetwork ===
+
+impl Networks {
+    pub fn find_match(&self, ip: IpAddr) -> Option<Cidr> {
+        let ip: Cidr = ip.into();
+        self.0
+            .iter()
+            .filter(|c| c.contains(&ip))
+            .min_by(|a, b| a.size().cmp(&b.size()))
+            .cloned()
+    }
+}
+
+// === impl UnmeshedNetwork ===
+
+impl From<linkerd_k8s_api::UnmeshedNetwork> for UnmeshedNetwork {
+    fn from(u: linkerd_k8s_api::UnmeshedNetwork) -> Self {
+        let name = u.name_unchecked();
+        let namespace = u
+            .namespace()
+            .expect("UnmeshedNetwork must have a namespace");
+
+        UnmeshedNetwork {
+            name,
+            namespace,
+            networks: Networks(u.spec.networks.clone()),
+            creation_timestamp: u.creation_timestamp().map(|d| d.0),
+        }
+    }
+}
+
+// === impl MatchedUnmeshedNetwork ===
+
+impl From<(Cidr, &UnmeshedNetwork)> for MatchedUnmeshedNetwork {
+    fn from((cidr, unet): (Cidr, &UnmeshedNetwork)) -> Self {
+        Self {
+            name: unet.name.clone(),
+            namespace: unet.namespace.clone(),
+            matched_cidr: MatchedCidr(cidr),
+            creation_timestamp: unet.creation_timestamp,
+        }
+    }
+}
+
+impl std::cmp::PartialOrd for MatchedUnmeshedNetwork {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl std::cmp::Ord for MatchedUnmeshedNetwork {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.matched_cidr
+            .cmp(&other.matched_cidr)
+            .then_with(|| self.creation_timestamp.cmp(&other.creation_timestamp))
+            .then_with(|| self.namespace.cmp(&other.namespace).reverse())
+            .then_with(|| self.name.cmp(&other.name).reverse())
+    }
+}
+
+// === impl MatchedCidr ===
+
+impl std::cmp::PartialOrd for MatchedCidr {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl std::cmp::Ord for MatchedCidr {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        let cidr_size_self = self.0.size();
+        let cidr_size_other = other.0.size();
+
+        match cidr_size_self.cmp(&cidr_size_other) {
+            std::cmp::Ordering::Less => std::cmp::Ordering::Greater,
+            std::cmp::Ordering::Greater => std::cmp::Ordering::Less,
+            std::cmp::Ordering::Equal => std::cmp::Ordering::Equal,
+        }
```

Comment:
we should add a comment here noting the ordering, i found the reversed `Less => Greater` and `Greater => Less` a little confusing.

alternately, we could `cidr_size_other.cmp(&cidr_size_self)` and achieve the same idea

Reply:
added more comments